pyparsing/core.py

```diff
@@ -3570,13 +3586,17 @@ def __init__(self):
     def preParse(self, instring: str, loc: int) -> int:
         if loc == 0:
             return loc
-        else:
-            ret = self.skipper.preParse(instring, loc)
-            if "\n" in self.orig_whiteChars:
-                while instring[ret : ret + 1] == "\n":
-                    ret = self.skipper.preParse(instring, ret + 1)
+
+        ret = self.skipper.preParse(instring, loc)
+
+        if not "\n" in self.orig_whiteChars:
```

Comment:
Rejected. I dislike the negative test here, and there is only a small bit of code being nested.